libs/wingc/src/type_check.rs

```diff
@@ -636,6 +636,28 @@ impl TypeRef {
 			_ => false,
 		}
 	}
+
+	pub fn is_serializable(&self) -> bool {
+		match **self {
+			Type::Resource(_) => false,
+			Type::Enum(_) => true,
+			Type::Number => true,
+			Type::String => true,
+			Type::Duration => true,
+			Type::Boolean => true,
+			Type::Json => true,
+			Type::Array(v) => v.is_serializable(),
+			Type::Map(v) => v.is_serializable(),
+			Type::Set(v) => v.is_serializable(),
+			Type::Struct(_) => self.as_struct().unwrap().env.iter(true).all(|(_, v, _)| {
+				v.as_variable()
+					.expect("Expected struct field to be a variable in the struct env")
+					.type_
+					.is_serializable()
+			}),
```

Comment:
As well lets keep structs false for now
```suggestion
			Type::Struct(_) => false,
```